src/index.ts

```diff
@@ -23,10 +23,48 @@ const getRepo = (ctx: Context) => {
 const repo = getRepo(context);
 
 setOutput("repo", `${repo.owner}/${repo.repo}`);
+const logger = generateCoreLogger();
 
-if (context.payload.pull_request) {
+const action = async () => {
   const token = getInput("GITHUB_TOKEN", { required: true });
-  const api = new PullRequestApi(getOctokit(token), generateCoreLogger());
-  const author = api.getPrAuthor(context.payload.pull_request as PullRequest);
-  info("Author of the PR is " + author);
-}
+  const repoInfo = getRepo(context);
+  const api = new PullRequestApi(getOctokit(token), repoInfo);
+  const prs = await api.listPRs(false);
+  if (prs.length > 0) {
+    logger.info(`About to update ${prs.length} PRs 🗄️`);
+    const rows: SummaryTableRow[] = [
+      [
+        { data: "PR Number", header: true },
+        { data: "Title", header: true },
+        { data: "Result", header: true },
+      ],
+    ];
+    for (const { number, title } of prs.sort((a, b) => a.number - b.number)) {
+      logger.info(`📡 - Updating '${title}' #${number}`);
+      const repoTxt = `${repoInfo.owner}/${repoInfo.repo}#${number}`;
+      try {
+        await api.update(number);
+        rows.push([repoTxt, title, "Pass ✅"]);
+        logger.info(`📥 - Updated #${number}`);
+      } catch (error) {
+        logger.error(error as string | Error);
+        rows.push([repoTxt, title, "Fail ❌"]);
+      }
+    }
+    logger.info("🪄 - Finished updating PRs");
+    await summary
+      .addHeading("Up to date", 1)
+      .addHeading("PRs updated", 3)
+      .addTable(rows)
+      .write();
+  } else {
+    logger.info("No matching PRs found. Aborting");
+    summary.addHeading("Up to date", 1).addHeading("No matching PRs found");
+  }
+};
+
+action()
+  .then(() => {
+    logger.info("Operation completed");
+  })
+  .catch((e) => logger.error(e as Error));
```

Comment:
Should we maybe `setFailed` here? Otherwise the action will be green even if it failed unexpectedly?